org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/wizard/project/GradleProjectWizardPage.java

```diff
@@ -99,6 +99,19 @@ private void bindToConfiguration() {
             public void modifyText(ModifyEvent e) {
                 File projectDir = FileUtils.getAbsoluteFile(GradleProjectWizardPage.this.projectDirText.getText()).orNull();
                 getConfiguration().setProjectDir(projectDir);
+                // TODO currently a
+                // com.gradleware.tooling.toolingutils.binding.ValidationListener
+                // only receives error messages, therefore the warning message
+                // is implemented here, but it should also be done by a
+                // com.gradleware.tooling.toolingutils.binding.Validator<T>
+                if (NONE == getMessageType()) {
+                    // in case the current message type is NONE, a warning may
+                    // be shown
+                    if (WorkspaceUtils.isInWorkspaceFolder(projectDir)) {
+                        setMessage("It is recommended to locate your Gradle projects outside the workspace location",
```

Comment:
I'd only state that for certain configurations the root project folder have to be renamed.